src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -46,23 +47,44 @@
 #include "runtime/handles.inline.hpp"
 #include "runtime/javaCalls.hpp"
 
-GrowableArray<char*>* LambdaFormInvokers::_lambdaform_lines = NULL;
+GrowableArrayCHeap<char*, mtClassShared>* LambdaFormInvokers::_lambdaform_lines = nullptr;
+Array<Array<char>*>* LambdaFormInvokers::_static_archive_invokers = nullptr;
+#define NUM_FILTER 4
+static const char* filter[NUM_FILTER] = {"java.lang.invoke.Invokers$Holder",
+                                         "java.lang.invoke.DirectMethodHandle$Holder",
+                                         "java.lang.invoke.DelegatingMethodHandle$Holder",
+                                         "java.lang.invoke.LambdaForm$Holder"};
 
 void LambdaFormInvokers::append(char* line) {
   if (_lambdaform_lines == NULL) {
-    _lambdaform_lines = new GrowableArray<char*>(100);
+    _lambdaform_lines = new GrowableArrayCHeap<char*, mtClassShared>(150);
   }
   _lambdaform_lines->append(line);
 }
 
+void LambdaFormInvokers::append_filtered(char* line) {
+  for (int k = 0; k < NUM_FILTER; k++) {
+    if (strstr(line, filter[k]) != nullptr) {
```

Comment:
Do you really want to match anywhere in the string? I'm assuming the line should start with the filter?

Reply:
You reminded me that only the invoker lines which contain four holder class be stored in static archive, not else. I will change that part. For the filtered lines, it starts with [LF_RESOLVE] .... so the comparison I used here to check if the line contains "java.lang.invoke.<Class$holder>".